pilot/pkg/model/service.go

```diff
@@ -1131,7 +1132,7 @@ func (s *Service) GetAddressForProxy(node *Proxy) string {
 // GetExtraAddressesForProxy returns a k8s service's extra addresses to the cluster where the node resides.
 // Especially for dual stack k8s service to get other IP family addresses.
 func (s *Service) GetExtraAddressesForProxy(node *Proxy) []string {
-	if node.Metadata != nil {
+	if features.EnableDualStack && node.Metadata != nil {
 		if node.Metadata.ClusterID != "" {
```

Comment:
**question:** is there a standard for where features are intended to be gated? This Get method falls in the middle of the chain. Does it make more sense to gate this at the terminal end closer to converting to envoy config (more like it works now) or at the initial conversion (in this case from k8s service into our model)?

Reply:
Reviewing where this method is used it looks to all be places you added it to support dual stack so it makes some sense to me to feature gate it here after all

Reply:
> **question:** is there a standard for where features are intended to be gated? This Get method falls in the middle of the chain. Does it make more sense to gate this at the terminal end closer to converting to envoy config (more like it works now) or at the initial conversion (in this case from k8s service into our model)?

Hi @ilrudie,  Yes, as a new feature in Istio, there should be flag to gate it if necessary. And yes, the change covered all places where you use the function `GetExtraAddressesForProxy`